src/libs/EmojiTrie.ts

```diff
@@ -81,37 +87,35 @@ function createTrie(lang: SupportedLanguage = CONST.LOCALES.DEFAULT): Trie<Emoji
     const defaultLangEmojis: LocalizedEmojis = localeEmojis[CONST.LOCALES.DEFAULT];
     const isDefaultLocale = lang === CONST.LOCALES.DEFAULT;
 
-    emojis.forEach((item: Emoji) => {
-        if (!item.name) {
-            return;
-        }
-
-        const englishName = item.name;
-        const localeName = langEmojis?.[item.code]?.name ?? englishName;
-
-        const node = trie.search(localeName);
-        if (!node) {
-            trie.add(localeName, {code: item.code, types: item.types, name: localeName, suggestions: []});
-        } else {
-            trie.update(localeName, {code: item.code, types: item.types, name: localeName, suggestions: node.metaData.suggestions});
-        }
-
-        const nameParts = getNameParts(localeName).slice(1); // We remove the first part because we already index the full name.
-        addKeywordsToTrie(trie, nameParts, item, localeName);
-
-        // Add keywords for both the locale language and English to enable users to search using either language.
-        const keywords = (langEmojis?.[item.code]?.keywords ?? []).concat(isDefaultLocale ? [] : defaultLangEmojis?.[item.code]?.keywords ?? []);
-        addKeywordsToTrie(trie, keywords, item, localeName);
-
-        /**
-         * If current language isn't the default, prepend the English name of the emoji in the suggestions as well.
-         * We do this because when the user types the english name of the emoji, we want to show the emoji in the suggestions before all the others.
-         */
-        if (!isDefaultLocale) {
-            const englishNameParts = getNameParts(englishName);
-            addKeywordsToTrie(trie, englishNameParts, item, localeName, true);
-        }
-    });
+    emojis
+        .filter((item: Emoji): item is SimpleEmoji => !(item as HeaderEmoji).header)
```

Comment:
Could you explain the purpose of this filter()?

Reply:
@fabioh8010 removing of the `if condition` and adding of this `filter` are related to each other.
`emojis` is an array that has one header emoji 
```
{
        header: true,
        icon: Smiley,
        code: 'smileysAndEmotion',
 }
```

and other usual emojis.

Header emoji is not used in emoji suggestions and was filtered out before with a check.
```
if (!item.name) {
    return;
}
```

I replaced it with a filter instead to add type narrowing to identify that only simple emojis are used in the suggestions.

Btw, before the migration of the EmojiTrie file it also checked for [the header field](https://github.com/Expensify/App/pull/27693/files#diff-e58f83faf68b131799a867a7057d6cd3aef2a834268541312e43c2f8e681da45L52) instead of name to filter header emoji out.

Reply:
Thanks for explanation @VickyStash !